tests/filecheck/transforms/dispatch_regions.mlir

```diff
@@ -113,4 +113,43 @@
 //CHECK-NEXT:   "func.func"() <{"sym_name" = "snax_is_dm_core", "function_type" = () -> i1, "sym_visibility" = "private"}> ({
 //CHECK-NEXT:   }) : () -> ()
 //CHECK-NEXT: }) : () -> ()
+// -----
+// test function with explicit linalg generic
+"builtin.module"() ({
+  "func.func"() <{"sym_name" = "simple_mult", "function_type" = (memref<64xi32>, memref<64xi32>, memref<64xi32>) -> (), "sym_visibility" = "public"}> ({
+  ^0(%0 : memref<64xi32>, %1 : memref<64xi32>, %2 : memref<64xi32>):
+    linalg.generic {
+        indexing_maps = [affine_map<(d0) -> (d0)>, 
+                         affine_map<(d0) -> (d0)>, 
+                         affine_map<(d0) -> (d0)>],
+        iterator_types = ["parallel"]
+    } ins(%0, %1 : memref<64xi32>, memref<64xi32>) outs(%2 : memref<64xi32>) {
+    ^bb0(%arg0: i32, %arg1: i32, %arg2: i32):
+      %3 = arith.muli %arg0, %arg1 : i32
+      linalg.yield %3 : i32
+  }
+    "func.return"() : () -> ()
+
+  }) : () -> ()
+}) : () -> ()
+
```

Comment:
how is this testing something different from the above test?

Reply:
the Dispatch_regions pass fails only when Linalg.generic is used in this way. when it is used in the other way (with quotation marks), the pass always performs correctly. we now test both cases

Reply:
That shouldn't be though, the quotation marks just mark generic vs. custom syntax of how the operation is printed. They should represent the exact same operation though...